### PR TITLE
KTOR-26 Remove text-encoding dependency

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-26](https://youtrack.jetbrains.com/issue/KTOR-26)

### CODE REVIEW

File: ktor-io/js/src/io/ktor/utils/io/js/TextDecoderFallback.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    override fun decode(): String = ""
+
+    override fun decode(buffer: ArrayBufferView): String = buildPacket {
+        val bytes = buffer as Int8Array
+        for (index in 0 until bytes.length) {
+            val byte = bytes[index]
+            val point: Int = byte.toCodePoint()
+
+            if (point < 0) {
+                check(!fatal) { "Invalid character: $point" }
```
Comment (cy6erGn0m) [inline ktor-io/js/src/io/ktor/utils/io/js/TextDecoderFallback.kt:56]: What exception should be produce here?

File: ktor-io/js/src/io/ktor/utils/io/js/TextDecoderFallback.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        check(ENCODING_ALIASES.contains(requestedEncoding)) { "$encoding is not supported." }
+    }
+
+    override fun decode(): String = ""
+
+    override fun decode(buffer: ArrayBufferView): String = buildPacket {
+        val bytes = buffer as Int8Array
+        for (index in 0 until bytes.length) {
+            val byte = bytes[index]
+            val point: Int = byte.toCodePoint()
```
Comment (cy6erGn0m) [inline ktor-io/js/src/io/ktor/utils/io/js/TextDecoderFallback.kt:53]: explicit type?

**Review by cy6erGn0m — APPROVED** [review]

File: ktor-io/js/src/io/ktor/utils/io/js/TextDecoderFallback.kt (hunk 1 above)
Comment (e5l) [reply ktor-io/js/src/io/ktor/utils/io/js/TextDecoderFallback.kt:56]: Negative chars don't belong to encoding

File: ktor-io/js/src/io/ktor/utils/io/js/TextDecoder.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.utils.io.js
+
+import org.khronos.webgl.*
+import kotlin.reflect.*
```
Comment (hfhbd) [inline ktor-io/js/src/io/ktor/utils/io/js/TextDecoder.kt:8]: Why do you need `reflect`?

Comment (e5l) [reply ktor-io/js/src/io/ktor/utils/io/js/TextDecoder.kt:8]: Thanks, will check

Comment (e5l) [reply ktor-io/js/src/io/ktor/utils/io/js/TextDecoder.kt]: Removed. IDE didn't show it as unused for some reason

**Pull request merged**